lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -98,6 +110,23 @@ spec = do
                     res <- try' (runGet getBlock bs)
                     res `shouldSatisfy` isRight
                     return ()
+        describe "whileM (not <$> isEmpty)" $ do
+            it "should fail immediately when the decoder errors" $ do
+                -- Context: We use whileM not <$> isEmpty to decode multiple
+                -- block-fragments/messages. If a message-decoder is wrong, we
+                -- want to know that clearly.
+                let getMessage' = label "getMessage" $ isolate 3 getWord16be
+                let getBlock' = whileM (not <$> isEmpty) getMessage'
+                res <- try' (runGet getBlock' $ BL.pack [0,0,0])
```

Comment:
slightly confusing to call these `getMessage` and `getBlock` since this test really has nothing to do with blocks and messages...